api/blueprints/drbCollection.py

```diff
@@ -252,3 +305,30 @@ def constructOPDSFeed(
     )
 
     return opdsFeed
+
+#Deleting the rows of collection_editions that were in the original collection
+def removeEditionsFromCollection(dbClient, collection):
+    dbClient.session.execute(COLLECTION_EDITIONS.delete().where(COLLECTION_EDITIONS.c.collection_id == collection.id))
+
+#Inserting rows of collection_editions based on editionIDs array
+def addEditionsToCollection(dbClient, collection, editionIDs):
+    dbClient.session.execute(COLLECTION_EDITIONS.insert().values([ \
+        {"collection_id": collection.id, "edition_id": eid} \
+        for eid in editionIDs \
+    ]))
+
+#Inserting rows of collection_editions based on workUUIDs array
+def addWorkEditionsToCollection(dbClient, collection, workUUIDs):
+    collectionWorks = dbClient.session.query(Work)\
+            .join(Work.editions)\
+            .filter(Work.uuid.in_(workUUIDs))\
+            .all()
+
+    for work in collectionWorks:
+        editions = [ed for ed in work.editions]
+        
+
+        dbClient.session.execute(COLLECTION_EDITIONS.insert().values([ \
+        {"collection_id": collection.id, "edition_id": edition.id} \
+        for edition in editions \
```

Comment:
I think we only want to insert one "preferred" edition into the collection per work. @sarangj I think you just implemented something similar for your most recent PR, would a similar approach work here?

Reply:
Yea, mine was kind of a gnarly SQL based solution: https://github.com/NYPL/drb-etl-pipeline/pull/225/files#diff-bb24eb4550556613d258592571b4d75aba40f31853c1b3fd384b886929b1d09eR59-R113

Here, we're not fetching as much data, so I think you could just sort the list of editions and pick the earliest one per collection?

Reply:
so like, i think if per work, you sort the list of editions by date create ascending and just use the first edition, that should do it.